pkg/providers/cloudstack/cloudstack.go

```diff
@@ -429,20 +444,20 @@ func (p *cloudstackProvider) validateSecretsUnchanged(ctx context.Context, clust
 }
 
 func secretDifferentFromProfile(secret *corev1.Secret, profile decoder.CloudStackProfileConfig) bool {
-	return string(secret.Data["uri"]) != profile.ManagementUrl ||
-		string(secret.Data["apikey"]) != profile.ApiKey ||
-		string(secret.Data["secretkey"]) != profile.SecretKey ||
-		string(secret.Data["verifyssl"]) != profile.VerifySsl
+	return string(secret.Data["api-url"]) != profile.ManagementUrl ||
```

Comment:
Why are we doing this? Seems unrelated to this PR

Reply:
I found this issue (secret data name) when testing upgrade test.